mm2src/coins/eth.rs

```diff
@@ -6179,3 +6190,755 @@ async fn get_eth_gas_details(
+                    let decoded = decode_contract_call(function, &tx_from_rpc.input.0)
+                        .map_to_mm(|err| ValidatePaymentError::TxDeserializationError(err.to_string()))?;
+
+                    let validation_params = ValidationParams {
+                        maker_address,
+                        etomic_swap_contract,
+                        token_id: args.token_id,
+                        amount: Some(args.amount.to_string()),
+                    };
+                    validate_decoded_data(&decoded, &validation_params)?;
+
+                    let taker_address =
+                        addr_from_raw_pubkey(args.taker_pub).map_to_mm(ValidatePaymentError::InternalError)?;
+                    let htlc_params = ExpectedHtlcParams {
+                        swap_id,
+                        taker_address,
+                        token_address,
+                        taker_secret_hash: args.taker_secret_hash.to_vec(),
+                        maker_secret_hash: args.maker_secret_hash.to_vec(),
+                        time_lock: U256::from(args.time_lock),
+                    };
+                    decode_and_validate_htlc_params(decoded, 4, htlc_params)?;
+                },
+                ContractType::Erc721 => {
+                    let function = self.erc721_transfer_with_data()?;
+                    let decoded = decode_contract_call(function, &tx_from_rpc.input.0)
+                        .map_to_mm(|err| ValidatePaymentError::TxDeserializationError(err.to_string()))?;
+
+                    let validation_params = ValidationParams {
+                        maker_address,
+                        etomic_swap_contract,
+                        token_id: args.token_id,
+                        amount: None,
+                    };
+                    validate_decoded_data(&decoded, &validation_params)?;
+
+                    let taker_address =
+                        addr_from_raw_pubkey(args.taker_pub).map_to_mm(ValidatePaymentError::InternalError)?;
+                    let htlc_params = ExpectedHtlcParams {
+                        swap_id,
+                        taker_address,
+                        token_address,
+                        taker_secret_hash: args.taker_secret_hash.to_vec(),
+                        maker_secret_hash: args.maker_secret_hash.to_vec(),
+                        time_lock: U256::from(args.time_lock),
+                    };
+                    decode_and_validate_htlc_params(decoded, 3, htlc_params)?;
+                },
+            },
+            EthCoinType::Eth | EthCoinType::Erc20 { .. } => {
+                return MmError::err(ValidatePaymentError::InternalError(
+                    "EthCoinType must be Nft".to_string(),
+                ))
+            },
+        }
+        Ok(())
+    }
+
+    async fn payment_status_v2(
+        &self,
+        swap_contract_addr: Address,
+        swap_id: Token,
+        contract: &Contract,
```

Comment:
Please rename this param to `swap_contract`

Reply:
fixed https://github.com/KomodoPlatform/komodo-defi-framework/pull/2084/commits/ba61d495b5db96f79948c6a7714713592f825ca8